### PR TITLE
[SEMI-MODULAR] CI Fix 2

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

We were missing a lot of icon states for various objects.
I've added them

## Why It's Good For The Game

missing icon states bad

## Changelog
:cl:
fix: A lot of icon states were missing
code: Electric Welders are much cleaner under the hood
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/bluespace_locker/code/game/objects/structures/crates_lockers/bluespace_locker.dm [hunk 1]
```diff
@@ -27,6 +27,8 @@
 	icon_state = null
 	desc = ""
 	cutting_tool = null
+	has_opened_overlay = FALSE
+	has_closed_overlay = FALSE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/bluespace_locker/code/game/objects/structures/crates_lockers/bluespace_locker.dm:31]: Does that have any impact on how the bluespace locker will look in-game?

Comment (ZephyrTFA) [reply modular_skyrat/modules/bluespace_locker/code/game/objects/structures/crates_lockers/bluespace_locker.dm:31]: It shouldnt due to the locker actively self changing icon state; that simply tells the base procs not to try and grab auto generatred icon states

File: code/modules/detectivework/detective_work.dm [hunk 2]
```diff
@@ -75,7 +75,7 @@
 
 /turf/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
 	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
-	if(!B)
+	if(!B || B.gc_destroyed) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```
Comment (Funce) [inline code/modules/detectivework/detective_work.dm]:
```suggestion
	if(QDELETED(X)) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```

File: code/modules/detectivework/detective_work.dm [hunk 3]
```diff
@@ -75,7 +75,7 @@
 
 /turf/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
 	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
-	if(!B)
+	if(QDELETED(X)) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```
Comment (ZephyrTFA) [inline code/modules/detectivework/detective_work.dm]:
```suggestion
	if(!B || QDELETED(B)) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```
Foxtrot how could u do this to me

File: code/modules/detectivework/detective_work.dm [hunk 4]
```diff
@@ -75,7 +75,7 @@
 
 /turf/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
 	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
-	if(!B)
+	if(!B || QDELETED(B)) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```
Comment (Funce) [inline code/modules/detectivework/detective_work.dm]:
```suggestion
	if(QDELETED(B)) // SKYRAT EDIT - SANITY CHECK; DONT ADD DNA TO A FUCKING DELETED BLOOD SPATTER - original: if(!B)
```
QDELETED includes the !B, pardon the typo on the previous~

Comment (ZephyrTFA) [reply code/modules/detectivework/detective_work.dm]: You owe me snuggles

**Pull request merged**